File: Python/0024_swap_nodes_in_pairs/0024_swap_nodes_in_pairs.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def swapPairs(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        dummy = ListNode(0)
        dummy.next = head
        pre = dummy
        while pre.next != None and pre.next.next != None:
            ListNode temp1 = pre.next
            ListNode temp2 = pre.next.next
            pre.next = temp2
            temp1.next = temp2.next
            temp2.next = temp1
            pre = pre.next.next   
        return dummy.next

File: Cplusplus/0024_swap_nodes_in_pairs/0024_swap_nodes_in_pairs.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */

class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        ListNode* dummy = new ListNode(0);
        dummy->next = head;
        ListNode* pre = dummy;
        while (pre->next != NULL && pre->next->next != NULL) {
            ListNode* temp1 = pre->next;
            ListNode* temp2 = pre->next->next;
            pre->next = temp2;
            temp1->next = temp2->next;
            temp2->next = temp1;
            pre = pre->next->next;
        }
        return dummy->next;
    }
};